src/enkf/gridio_fv3reg.f90

```diff
@@ -659,496 +661,1825 @@ subroutine writegriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,vargrid
+                     & k, minval(vargrid(:,k,nb,ne)), maxval(vargrid(:,k,nb,ne))
+          enddo
+
+          deallocate(uworkvar3d)
+       endif !iope == 0
+
+      endif
+
+      if (v_ind > 0) then
+        if(iope == 0 ) then 
+            allocate(vworkvar3d(nx_res+1,ny_res,nlevs))
+        else
+            allocate(vworkvar3d(1,1,nlevs))
+        endif
+        allocate(locvworkvar3d(nxloc+1,nyloc,nlevs))
+        varstrname = 'v'
+        call fv3lamfile%get_idfn(varstrname,file_id,fv3filename)
+        call read_fv3_restart_data3d(varstrname,fv3filename,file_id,locvworkvar3d)
         do k=1,nlevs
+          call mpi_gatherv(locvworkvar3d(1:nxloc,1:nyloc,k), recvcounts2d(iope+1), mpi_real4, &
+                     workvar2d, recvcounts2d, displs2d,&
+                     mpi_real4, 0,iocomms(mem_pe(nproc)) ,iret)
+          if(iope==0)  vworkvar3d(1:nx_res,1:ny_res,k)=workvar2d
+        enddo
+        if(iope==0) then
+         do k=1,nlevs
            nn = nn_tile0
            do j=1,ny_res
-              do i=1,nx_res
-                 nn=nn+1
-                 workinc3d(i,j,k)=vargrid(nn,levels(q_ind-1)+k,nb,ne) 
-              enddo
+             do i=1,nx_res
+                nn=nn+1
+                vargrid(nn,levels(v_ind-1)+k,nb,ne)=vworkvar3d(i,j,k) 
+             enddo
            enddo
+         enddo
+         do k = levels(v_ind-1)+1, levels(v_ind)
+           if (nproc .eq. 0)                                               &
+                 write(6,*) 'READFVregional : v ',                           &
+                     & k, minval(vargrid(:,k,nb,ne)), maxval(vargrid(:,k,nb,ne))
+         enddo
+         deallocate(vworkvar3d)
+        endif !iope ==0 
+     endif !v_ind
+
+     if (w_ind > 0) then
+        varstrname = 'W'
+        call fv3lamfile%get_idfn(varstrname,file_id,fv3filename)
+        call read_fv3_restart_data3d(varstrname,fv3filename,file_id,locworkvar3d)
+        do k=1,nlevs
+           call mpi_gatherv(locworkvar3d(1:nxloc,1:nyloc,k), recvcounts2d(iope+1), &
+                       mpi_real4, workvar3d(:,:,k), recvcounts2d, displs2d,&
+                       mpi_real4, 0,iocomms(mem_pe(nproc)) ,iret)
         enddo
-       qworkvar3d=qworkvar3d+workinc3d   
+        if(iope==0) then
+          do k=1,nlevs
+             nn = nn_tile0
+             do j=1,ny_res
+               do i=1,nx_res
+                  nn=nn+1
+                  vargrid(nn,levels(w_ind-1)+k,nb,ne)=workvar3d(i,j,k)
+               enddo
+             enddo
+          enddo
+          do k = levels(w_ind-1)+1, levels(w_ind)
+            if (nproc .eq. 0)                                               &
+                   write(6,*) 'READFVregional : w ',                           &
+                       & k, minval(vargrid(:,k,nb,ne)), maxval(vargrid(:,k,nb,ne))
+          enddo
+        endif !iope == 0 
+
+    endif
+
+    if (tv_ind > 0.or.tsen_ind) then
```

Comment:
Please add > 0 to tsen_ind

Reply:
Done! Thanks for pointing out this issue